### PR TITLE
syslog-ng: 3.6.2 -> 3.9.1

### LINKED ISSUE #20153: systemd syslog-ng service fails to activate
I'm upgrading my systems from NixOS 16.03 to 16.09 and I'm trying to get syslog-ng working again. It's currently configured as:

```nix
{ syslog-ng.enable = true; }
```

The issue is that systemd is not noticing that syslog-ng has started:

```
Nov 04 12:47:08 mymachine systemd[1]: syslog-ng.service: Start operation timed out. Terminating.
Nov 04 12:47:08 mymachine systemd[1]: Failed to start syslog-ng daemon.
Nov 04 12:47:08 mymachine systemd[1]: syslog-ng.service: Unit entered failed state.
Nov 04 12:47:08 mymachine systemd[1]: syslog-ng.service: Failed with result 'timeout'.
Nov 04 12:47:08 mymachine systemd[1]: syslog-ng.service: Service hold-off time over, scheduling restart.
```

I do see that syslog-ng is running:

```
# ps -AH a | grep syslog
/nix/store/kag473gb3w8zaqw09qkgsjg2qn4ra5zc-syslog-ng-3.6.2/sbin/syslog-ng --foreground --module-path=/nix/store/kag473gb3w8zaqw09qkgsjg2qn4ra5zc-syslog-ng-3.6.2/lib/syslog-ng --cfgfile=/nix/store/8f9npiv1fyc487haxj0ff957ag7zqzld-syslog-ng.conf --control=/run/syslog-ng/syslog-ng.ctl --persist-file=/var/syslog-ng/syslog-ng.persist --pidfile=/run/syslog-ng/syslog-ng.pid
```

Note that the systemd syslog-ng service is [configured](https://github.com/NixOS/nixpkgs/blob/release-16.09/nixos/modules/services/logging/syslog-ng.nix#L87) with `Type = "notify";` This [means](https://www.freedesktop.org/software/systemd/man/systemd.service.html#Type=) that syslog-ng should notify systemd when it has started up. My current hypotheses is that with the systemd upgrade to 231 something in the notification protocol has changed causing the syslog-ng notification to fail.

### PR DESCRIPTION
**Pull request by basvandijk**
@fpletz this fixes https://github.com/NixOS/nixpkgs/issues/20153.

Since syslog-ng is currently broken on 16.09 this should be cherry-picked on `release-16.09`.

I tested it with the following configuration:

```nix
{
  services.syslog-ng = {
    enable = true;
    extraConfig = ''
      source s_network_tcp {
        network (
          ip("0.0.0.0")
          port(601)
          transport("tcp")
        );
      };
     
      destination d_devlog {
        unix-dgram(
          /dev/log
          flags(syslog-protocol)
        );
      };
     
      log {
        source(s_network_tcp);
        destination(d_devlog);
      };
    '';
  };
}
```

If I then log a message using:

```
logger --server=localhost --tcp "Hello World"
```

I see the following in my `journalctl`:

```
Feb 13 18:07:15 bas.van.dijk.lumi syslog-ng[13387]: 1 2017-02-13T18:07:15+01:00 localhost 1 - - - 2017-02-13T18:07:15.046465+01:00 bas.van.dijk.lumi bas.van.dijk - - [timeQuality tzKnown="1" isSynced="1" syncAccuracy="229387"] Hello World
```

### CODE REVIEW

**Comment by grahamc** [comment]
In what way is syslog-ng broken on 16.09?

**Comment by basvandijk** [comment]
@grahamc see https://github.com/NixOS/nixpkgs/issues/20153.

File: pkgs/tools/system/syslog-ng/default.nix [hunk 1]
```diff
@@ -1,17 +1,39 @@
-{ stdenv, fetchurl, eventlog, pkgconfig, glib, python, systemd, perl
-, riemann_c_client, protobufc, pcre, yacc }:
+{ stdenv, fetchgit, autoconf, autoconf-archive, automake, libtool, flex, openssl
+, eventlog, pkgconfig, glib, python, systemd, perl
+, riemann_c_client, protobufc, pcre, yacc, which }:
 
-stdenv.mkDerivation rec {
-  name = "syslog-ng-${version}";
+stdenv.mkDerivation {
+  name = "syslog-ng-3.9.1";
```
Comment (abbradar) [inline pkgs/tools/system/syslog-ng/default.nix]: Please leave `version`. It was discussed (admittedly a long time ago and it'd be difficult to find) that we might move to using it insttead of parsing `name` some day; it'd definitely be easier to move if most packages already contain it. Disclosure: I want it to happen very much as current way seems ugly to me ^_^

Comment (basvandijk) [inline pkgs/tools/system/syslog-ng/default.nix]: Sure no problem.

**Comment by basvandijk** [comment]
@fpletz thanks for the refactoring. It looks good!

Since I'm now using a local override of syslog-ng I don't care that much anymore if this gets cherry-picked on 16.09. Waiting on 17.03 is fine by me.

**Pull request merged**